packages/ipfs-core/src/components/bitswap/unwant.js

```diff
@@ -27,8 +24,14 @@ module.exports = ({ bitswap }) => {
    * console.log(list)
    * // []
    * ```
+   *
+   * @param {CID | CID[]} cids - The CIDs to remove from the wantlist
+   * @param {AbortOptions} [options]
```

Comment:
I'm not sure I fully understand your comment here. In most places in the js-ipfs code base we accept more options we do something along the lines of:

```js
/**
 * @param {UnwantOptions & AbortOptions} [options]
 */
```
Here we do not have any other options so defining `UnwantOptions` made no sense. Once we introduce more options we will define such `UnwantOptions` and update param annotation as shown above.

If we just type it as `Object` than type check will pretty much treat it as any, meaning it would not show anything useful in docs nor it will be able to point out problems e.g. when some option is expected but not provided.

> I would keep this as an object and would add the properties of AbortOptions below

If you mean something like this:

```js
@param {Object} [options]
@param {number|string] [options.timer]
@param {AbortSignal] [options.signal]
```
We could do that, but that gets very tedious really soon as pretty much every function takes these, which is why we settled on `{MyOptions & AbortOptions}` instead.

Reply:
`{MyOptions & AbortOptions}` makes sense! I did see it in other examples. We will need to change them in the future if we need other options, but I understand your point.